### DISCUSSION TITLE
using other webcore applications from a  master app

### DISCUSSION THREAD

**Discussion by RamazanGeven** [500Foods/TMS-WEB-Core-TemplateDemo#1] [Ideas]
hi Andrew , 
i want to use your template for a project .
ı am not so experience on TMS yet.. but can learn fast.
i used tms datamodeler , your examples , and when i think at this point need this:
i will add a pagecontrol , pages and TWebBrowserControl on everytab
when user click on menu then i will call another webcore application in this WebBrowserControl but need send startup paremeters (jwt , user_id , time period or code for tenant etc)
what do you think about this?

**Comment by 500Foods** [comment]
I don't quite follow what you're trying to do?

The way this is constructed now, once you get past the login form, additional forms are used to present the rest of the user interface.  If you have additional TMS WEB Core applications that you've written yourself, it would be best to just incorporate them into forms directly.  You wouldn't necessarily want to launch another TMS WEB Core app within this one.

**Reply by RamazanGeven** [reply]
Then js file will be big. and everytime need compile if i add something. 
i want a kind of moduler system .
If a module become stable then will no need compile again.

**Comment by 500Foods** [comment]
Couple of ideas.
1. The default for TMS WEB Core projects is to produce a single .js file but there is an option to compile into separate .js files, so if nothing changes for a particular form, then perhaps the .js file for that form won't change.
2. You certainly can embed other projects, websites and that sort of thing using <iframe> components in particular.  If they're running from the same domain, you might be able to use localStorage to pass data back and forth, like JWTs and other parameters.

**Comment by RamazanGeven** [comment]
3- maybe can use send parameters with url 
4-maybe can use a websocket server for communication with all modules 
5- ITMSFNCCustomWebBrowserBridge. ;  i used it between exe and webcore application. it s very good but still i can not use it in webcore app and with a TWebBrowserControl .  
i was busy other projects  and alwyas i wanted free time for start TMS .. now it s possiable for me. So i will bother you often if you agree

**Comment by 500Foods** [comment]
3 - if you use an <iframe> element, then yes you can pass parameters into the application at startup easily enough.  It is not very easy to pass data back the other way though.  
4  - WebSockets might work.  But if there is a lot of data to pass back and forth regularly, finding a way to integrate these other TMS WEB Core apps into one system might still be a better idea. 
5 - Same here - if you're running multiple apps from the same domain, and they're apps you've created yourself, the "modularity" benefits are perhaps less important than having an app to manage things better overall - which is one of the reasons this template was created in the first place.  Modularity at the form level.

I'm happy to answer any questions.  If you send too many it will just take longer for me (or hopefully others as well) to answer them 😄   Be sure to read through the TMS blog posts about this template project and the other one, and maybe some of the other blog posts I've written as well if you want examples of other JS libraries.
